exercises/Histogram/histogram.py

```diff
@@ -0,0 +1,41 @@
+
+def read_file(filename):
+    with open(filename) as f:
+        line = f.readlines()
+    n = line[0].strip("\n").split(" ")
+    a = line[1].strip("\n").split(" ")
+    return a, n
+
+
+def histogram(a, n):
+    maximum = max(float(x) for x in a)
+    minimum = min(float(x) for x in a)
+    interval = float(maximum - minimum)
+    scale = interval / float(n[0])
+    bins = []
+    intervals_dict = {}
+    x = minimum
+    while x <= maximum:
+        bins.append(x)
+        x += scale
+    for i in range(1, len(bins)):
+        intervals_dict.setdefault(bins[i], 0)
+    for x in a:
+        for i in range(1, int(n[0]) + 1):
```

Comment:
There is no need for the second loop, you can calculate the bin number directly:
`bin_number=min(floor((x-minimum)/scale),n-1)`